### PR TITLE
Custom deployment policy

### LINKED ISSUE #140: DeploymentPolicy is not fully configurable
Have a question? Please checkout our [Slack Community](https://slack.cloudposse.com) or visit our [Slack Archive](https://archive.sweetops.com/). 

[Slack Community](https://slack.cloudposse.com)

## Describe the Feature

DeploymentPolicy can currently only be `Immutable` or `Rolling`, but I would like to use others like `RollingWithAdditionalBatch`.

## Expected Behavior

There would be a `deployment_policy` parameter that can be used to specify required deployment policy and it would support all policies: All at once, Rolling, Rolling with additional batch, Immutable,Traffic splitting

## Alternatives Considered

Tried to use `additional_settings` to define the unsupported policy but the original one applies instead.

### PR DESCRIPTION
**Pull request by v3rm0n**
## what
Added `deployment_policy` variable.

## why
So that other deployment policies, in addition to Immutable and Rolling, can be used.

## references
* [All supported policies](https://docs.aws.amazon.com/elasticbeanstalk/latest/dg/using-features.rolling-version-deploy.html)
* Closes #140

### CODE REVIEW

**Comment by rennokki** [comment]
@v3rm0n Rebase? @htplbc @aknysh bump, same problem here

**Comment by rennokki** [comment]
Not being able to plan without having a false positive is painful when you need RollingWithAdditionalBatch. Any update on this? @htplbc @aknysh Can this be merged?

**Comment by aknysh** [comment]
@v3rm0n  please resolve the conflicts

**Review by jbonnier — APPROVED** [review]

**Comment by rennokki** [comment]
Do you need maintainers for this package? I'll be happy to do it. 😅

**Comment by stamtech** [comment]
bump, same problem here, will this be merged soon ?

File: main.tf [hunk 1]
```diff
@@ -726,7 +726,7 @@ resource "aws_elastic_beanstalk_environment" "default" {
   setting {
     namespace = "aws:elasticbeanstalk:command"
     name      = "DeploymentPolicy"
-    value     = var.rolling_update_type == "Immutable" ? "Immutable" : "Rolling"
+    value     = var.deployment_policy != "" ? var.deployment_policy : (var.rolling_update_type == "Immutable" ? "Immutable" : "Rolling")
```
Comment (aknysh) [inline main.tf:729]:
```suggestion
    value     = var.deployment_policy
```

why do we need to mix `rolling_update_type` with `deployment_policy`?
Those are diff concepts not related to each other
It was incorrectly done before, let's fix it now.

**Review by aknysh — COMMENTED** [review]
@v3rm0n  thanks, please see comments

Please also run

```
make init
make github/init
make readme
```

and commit all the changes to README and GitHub actions

Thanks

**Comment by aknysh** [comment]
to speed up the review process, you can join our Slack community https://slack.sweetops.com/ and ping us directly

File: main.tf (hunk 1 above)
Comment (v3rm0n) [reply main.tf:729]: It was just for backwards compatibility, but yeah they are different, will change it

**Review by aknysh — APPROVED** [review]
thanks @v3rm0n

**Pull request merged**